addons/website_sale/controllers/main.py

```diff
@@ -576,12 +576,30 @@ def checkout_form_validate(self, mode, all_form_values, data):
         error = dict()
         error_message = []
 
-        # prevent name change if invoices exist
         if data.get('partner_id'):
-            partner = request.env['res.partner'].browse(int(data['partner_id']))
-            if partner.exists() and partner.sudo().name and not partner.sudo().can_edit_vat() and 'name' in data and (data['name'] or False) != (partner.sudo().name or False):
+            partner_su = request.env['res.partner'].sudo().browse(int(data['partner_id'])).exists()
```

Comment:
partner.exists() was done without sudo before. Seems like your refactoring change this part. Maybe the same, but one more risk in stable due to the refactoring.